src/pybind/mgr/cephadm/ssh.py

```diff
@@ -0,0 +1,257 @@
+import logging
+import os
+import asyncio
+from tempfile import NamedTemporaryFile
+from threading import Thread
+from contextlib import contextmanager
+from io import StringIO
+from shlex import quote
+from typing import TYPE_CHECKING, Optional, List, Tuple, Dict, Any, Iterator
+from orchestrator import OrchestratorError
+
+try:
+    import asyncssh
+except ImportError:
+    asyncssh = None
+
+if TYPE_CHECKING:
+    from cephadm.module import CephadmOrchestrator
+    from asyncssh.connection import SSHClientConnection
+
+logger = logging.getLogger(__name__)
+
+asyncssh_logger = logging.getLogger('asyncssh')
+asyncssh_logger.propagate = False
+
+DEFAULT_SSH_CONFIG = """
+Host *
+  User root
+  StrictHostKeyChecking no
+  UserKnownHostsFile /dev/null
+  ConnectTimeout=30
```

Comment:
shall we add 
```
    GSSAPIAuthentication yes
```

to match the settings on `/etc/ssh/ssh_config` on sepia? since we have this setting system-wide. would be better to have it when connecting to test nodes. as per https://asyncssh.readthedocs.io/en/latest/api.html#supported-server-config-options, `GSSAPIAuthentication` is supported.

Reply:
I closed https://github.com/ceph/ceph-cm-ansible/pull/646 because of https://github.com/ceph/ceph/pull/42051#issuecomment-903036575 I think the issue was that asyncssh was attempting GSSAPI auth before publickey auth and failing somewhere, so I think it's best if we avoid enabling GSSAPI for cephadm's usage of asyncssh. It looks like tests are passing now when I remove GSSAPI auth from the list of preferred methods for asyncssh. I haven't tested what would happen if someone passed in a config that enables GSSAPI auth, though. I imagine it would either do nothing (since we use publickey) or break it